### DISCUSSION TITLE
Set mouseButton.left to CameraControls.ACTION.TRUCK in React Three Fiber

### DISCUSSION THREAD

**Discussion by dayunbao** [yomotsu/camera-controls#501] [Q&A] [answered]
Hi,

I'm building a comic book creation app, and I recently started porting code that I originally wrote in a game engine to React (Next.js).  I plan on using Three.js (via React Three Fiber) to handle some of the 2D and 3D stuff the game engine was previously responsible for.  However, I've run into a problem.  I want to assign "truck" to the left mouse button instead of rotation, since I only want to be able to pan the camera and not rotate it.  The camera will be looking at a plane (the comic book page) and shouldn't be able to rotate around it.  I only want pan and zoom to be controlled by the mouse.  I think I should be able to set this via the props, but I can't find the `mouseButtonAction` to import, or the proper way to set it in React Three Fiber.  See the screenshot.

Does anyone know how I should do this?

Thanks!

![CameraControls_error](https://github.com/yomotsu/camera-controls/assets/11196050/2a6e1df1-001f-4509-a6c2-88a726494c4a)

**Comment by dayunbao** [comment, marked as answer]
Never mind, after looking a bit more, there are ultimately just `number`s.  So I used the values directly.

```
export declare const ACTION: Readonly<{
    readonly NONE: 0;
    readonly ROTATE: 1;
    readonly TRUCK: 2;
    readonly OFFSET: 4;
    readonly DOLLY: 8;
    readonly ZOOM: 16;
```

**Comment by abernier** [comment]
this is new https://drei.docs.pmnd.rs/controls/camera-controls#inputs
